src/components/hooks/useChangeServerLocale.ts

```diff
@@ -0,0 +1,46 @@
+import { updateUsers } from "api/users";
+import { RealmContext } from "providers/contexts.ts";
+import { useEffect, useState } from "react";
+import Queue from "realmModels/Queue.ts";
+import {
+  useAuthenticatedMutation
+} from "sharedHooks";
+
+const { useRealm } = RealmContext;
+
+const useChangeServerLocale = ( ) => {
+  const realm = useRealm( );
+  const [isMutating, setIsMutating] = useState( false );
+
+  const queue = realm.objects( "Queue" ).sorted( "id" );
+  const firstQueueItem = queue?.[0];
+  const mutationParams = firstQueueItem
+    ? JSON.parse( firstQueueItem?.payload )
+    : null;
```

Comment:
Figuring out which item is first in the queue and rehydrating its JSON string into an object belongs in the `dequeue()` method, because *every* consumer that wants to get something out of the queue will have to do that, and the FIFO processing order is the very definition of the queue so that needs to be internal. If a consumer can get the *last* item in the queue, then it's not a queue.

Since this is a work queue and you probably only want one worker working at a time, you might want to take a different approach where the worker claims the item at the head of the queue so no one else can work on it *but* doesn't remove it, when other potential workers peek at the head of the queue, they know not to work on it. When the worker is done working on the queue item, then it deletes it from the queue.